.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,754 @@
+      [mode: ${{
+        fromJSON(needs.pre-setup.outputs.is-untagged-devel)
+        && 'nightly' || ''
+      }}${{
+        fromJSON(needs.pre-setup.outputs.release-requested)
+        && 'release' || ''
+      }}${{
+        (
+          !fromJSON(needs.pre-setup.outputs.is-untagged-devel)
+          && !fromJSON(needs.pre-setup.outputs.release-requested)
+        ) && 'test' || ''
+      }}]
+    needs:
+    - pre-setup  # transitive, for accessing settings
+
+    runs-on: ubuntu-latest
+
+    env:
+      PY_COLORS: 1
+
+    steps:
+    - name: Switch to using Python v3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: >-
+          3.10
+    - name: >-
+        Calculate Python interpreter version hash value
+        for use in the cache key
+      id: calc-cache-key-py
+      run: |
+        from hashlib import sha512
+        from sys import version
+        hash = sha512(version.encode()).hexdigest()
+        print(f'::set-output name=py-hash-key::{hash}')
+      shell: python
+    - name: Get pip cache dir
+      id: pip-cache-dir
+      run: >-
+        echo "::set-output name=dir::$(python -m pip cache dir)"
+    - name: Set up pip cache
+      uses: actions/cache@v2.1.7
+      with:
+        path: ${{ steps.pip-cache-dir.outputs.dir }}
+        key: >-
+          ${{ runner.os }}-pip-${{
+          steps.calc-cache-key-py.outputs.py-hash-key }}-${{
+          needs.pre-setup.outputs.cache-key-files }}
+        restore-keys: |
+          ${{ runner.os }}-pip-${{
+              steps.calc-cache-key-py.outputs.py-hash-key
+          }}-
+          ${{ runner.os }}-pip-
+    - name: Install build tools
+      run: >-
+        python -m
+        pip install
+        --user
+        --upgrade
+        build
+
+    - name: Grab the source from Git
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: >-
+          ${{
+              steps.request-check.outputs.release-requested == 'true'
+              && 1 || 0
+          }}
+        ref: ${{ github.event.inputs.release-commitish }}
+
+    - name: Setup git user as [bot]
+      if: >-
+        fromJSON(needs.pre-setup.outputs.is-untagged-devel)
+        || fromJSON(needs.pre-setup.outputs.release-requested)
+      uses: fregante/setup-git-user@6cef8bf084d00360a293e0cc3c56e1b45d6502b8
+    - name: >-
+        Tag the release in the local Git repo
+        as ${{ needs.pre-setup.outputs.git-tag }}
+        for setuptools-scm to set the desired version
+      if: >-
+        fromJSON(needs.pre-setup.outputs.is-untagged-devel)
+        || fromJSON(needs.pre-setup.outputs.release-requested)
+      run: >-
+        git tag
+        -m '${{ needs.pre-setup.outputs.git-tag }}'
+        '${{ needs.pre-setup.outputs.git-tag }}'
+        --
+        ${{ github.event.inputs.release-commitish }}
+    - name: Build dists
+      run: >-
+        python
+        -m
+        build
+        --sdist
+        --wheel
```

Comment:
It's best not to have these args explicit. If they are, both the sdist and the wheel are created from Git checkout. But if you omit them, then the wheel is created from the sdist which also makes sure to test how pip builds wheels.
```suggestion
```

Reply:
thanks, didn't realize it affected wheel creation